langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -1114,6 +1114,16 @@ function testAsyncFpArgsWithArrays() returns [int, int[]] {
     return [reduce, filter];
 }
 
+function testReadOnlyArrayFilter() {
+    int[] & readonly numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    int[] evenNumbers = numbers.filter(val => val % 2 == 0);
+    int count = 0;
+    foreach int number in evenNumbers {
+        assertValueEquality(count * 2, number);
+        count += 1;
+    }
```

Comment:
Shall we also assert that the new value is not immutable? Please add for the map, table, and record tests too.
```suggestion
    }
    assertFalse(evenNumbers.isReadOnly());
```